roseus_smach/src/state-machine-ros.l

```diff
@@ -141,6 +141,28 @@
      ;; TODO apply userdata
      (setq init-tm (ros::time-now))
      ))
+  (:register-exit-signal-hook
+    (&optional es userdata exit-signal-hook-func)
+    (if (null es) (return-from :exit-state exit-state))
+    (if (null (derivedp es state))
+      (setq es (send self :state-machine :node es)))
+    ;; update exit-state slot
+    (setq exit-state es)
+    (send self :put 'userdata userdata)
+    (let ((exit-signal-hook
+            `(lambda-closure nil 0 0 (sig code)
+               (funcall ',exit-signal-hook-func (send ,self :get 'userdata))
+               (send ,self :exit-signal-hook sig code ,es (send ,self :get 'userdata)))))
+      (unix:signal unix::sigint exit-signal-hook)))
+  (:exit-signal-hook (sig code &optional es userdata)
+    (if (null es) (return-from :exit-signal-hook (exit sig)))
+    (ros::ros-error "exit-signal-hook called.~%")
+    (ros::ros-warn " Called ~A~%" (unix:signal unix::sigint))
```

Comment:
I would prefer to have this warning near the point where the function is actually called. Possibly before it is called (instead of after) for better debugging purposes.

Reply:
I updated.